contracts/utils/convex/ConvexStakingWrapper.sol

```diff
@@ -100,23 +99,40 @@ contract ConvexStakingWrapper is ERC20 {
     function addRewards() public {
         address mainPool = convexPool;
 
-        uint256 rewardsLength = rewards.length;
-
-        if (rewardsLength == 0) {
+        if (rewards.length == 0) {
             RewardType storage reward = rewards.push();
             reward.reward_token = crv;
             reward.reward_pool = mainPool;
-            rewardsLength = 1;
+            reward.reward_integral = 0;
+            reward.reward_remaining = 0;
+
+            reward = rewards.push();
+            reward.reward_token = cvx;
+            reward.reward_pool = address(0); // This is set to address(0) as initially we don't know if the pool has cvx rewards
+            reward.reward_integral = 0;
+            reward.reward_remaining = 0;
+
+            registeredRewards[crv] = CRV_INDEX + 1; //mark registered at index+1
+            registeredRewards[cvx] = CVX_INDEX + 1; //mark registered at index+1
         }
 
         uint256 extraCount = IRewardStaking(mainPool).extraRewardsLength();
-
-        uint256 startIndex = rewardsLength - 1;
-        for (uint256 i = startIndex; i < extraCount; i++) {
+        for (uint256 i = 0; i < extraCount; i++) {
             address extraPool = IRewardStaking(mainPool).extraRewards(i);
-            RewardType storage reward = rewards.push();
-            reward.reward_token = IRewardStaking(extraPool).rewardToken();
-            reward.reward_pool = extraPool;
+            address extraToken = IRewardStaking(extraPool).rewardToken();
+            if (extraToken == cvx) {
+                //update cvx reward pool address
+                rewards[CVX_INDEX].reward_pool = extraPool;
+            } else if (registeredRewards[extraToken] == 0) {
```

Comment:
This means that the CVX pool can be updated, but not the pool for other tokens. That is inconsistent.